Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/categories-tree/CategoriesTree.tsx

```diff
@@ -0,0 +1,105 @@
+﻿import React, { useCallback, useEffect, useState } from 'react';
+import { TreeItem, TreeView } from '@material-ui/lab';
+import { isArray, isEmpty, without } from 'lodash';
+import MdExpandMoreIcon from '../../guidelines/icons/MdExpandMoreIcon';
+import MdChevronRightIcon from '../../guidelines/icons/MdChervronRightIcon';
+import ITreeItemType from '../../../common/tree-types';
+import { ITreeProps } from '../../guidelines/trees/Tree';
+
+import styles from './CategoriesTree.module.scss';
+
+interface ICategoriesTreeProps extends ITreeProps{
+}
+
+const CategoryTree = ({
+    items,
+    onCategoryLabelClick,
+    defaultSelected,
+    defaultExpanded = [],
+}: ICategoriesTreeProps) => {
+    const [ expandedCategoriesIds, setExpanded ] = useState([] as string[]);
+    const [ selectedCategoriesIds, setSelected ] = useState('');
+
+    const handleTreeItemClick = useCallback(
+        (node: ITreeItemType) => {
+            const id = node.id.toString();
+            const newExpanded = expandedCategoriesIds.includes(id)
+                ? without(expandedCategoriesIds,id)
+                : [ ...expandedCategoriesIds,id ];
+
+            setExpanded(newExpanded);
+        },
+        [ expandedCategoriesIds, setExpanded ],
+    );
+
+    const handleLabelClick = useCallback(
+        (node: ITreeItemType) => {
+            setSelected(node.id.toString());
+
+            onCategoryLabelClick(node);
+        },
+        [ onCategoryLabelClick ],
+    );
+
+    useEffect(
+        () => {
+            if (isEmpty(selectedCategoriesIds) && defaultSelected) {
+                setSelected(defaultSelected);
+            }
+        },
+        [ defaultSelected, selectedCategoriesIds ],
+    );
+
+    useEffect(
+        () => {
+            if (isEmpty(expandedCategoriesIds) && !isEmpty(defaultExpanded)) {
+                setExpanded(defaultExpanded);
+            }
+        },
+        [ defaultExpanded, expandedCategoriesIds ],
+    );
```

Comment:
These can be removed, if initial state for `expandedCategoriesIds` and `defaultSelected` is set